src/schema/schema_create.c

```diff
@@ -961,6 +977,91 @@ __create_data_source(
     return (dsrc->create(dsrc, &session->iface, uri, (WT_CONFIG_ARG *)cfg));
 }
 
+/*
+ * __create_import_cmp --
+ *     Qsort function: sort the import entries array by name.
+ */
+static int WT_CDECL
+__create_import_cmp(const void *a, const void *b)
+{
+    WT_IMPORT_ENTRY *ae, *be;
+
+    ae = (WT_IMPORT_ENTRY *)a;
+    be = (WT_IMPORT_ENTRY *)b;
+
+    return strcmp(ae->name, be->name);
+}
+
+/*
+ * __create_meta_entry_worker --
+ *     Worker function for metadata file reader procedure. The function populates the import list
+ *     with entries related to the import URI.
+ */
+static int
+__create_meta_entry_worker(WT_SESSION_IMPL *session, WT_ITEM *key, WT_ITEM *value, void *state)
+{
+    WT_IMPORT_LIST *import_list;
+    const char *meta_key, *meta_key_suffix, *meta_value;
+
+    import_list = (WT_IMPORT_LIST *)state;
+    meta_key = (const char *)key->data;
+    meta_value = (const char *)value->data;
+
+    /* Get suffix of the key. */
+    meta_key_suffix = strchr(meta_key, ':');
+    WT_ASSERT(session, meta_key_suffix != NULL && meta_key_suffix[1] != '\0');
+    ++meta_key_suffix;
+
+    /* Skip unrelated entries. */
+    if (!WT_PREFIX_MATCH(meta_key_suffix, import_list->uri_suffix))
+        return (0);
+
+    /*
+     * We are not checking if the entry already exists in the metadata. It will be handled later in
+     * the appropriate create call.
+     */
+
+    /* Grow the entries array if needed. */
+    WT_RET(__wt_realloc_def(session, &import_list->entries_allocated, import_list->entries_next + 1,
+      &import_list->entries));
+
+    /* Populate the next entry. */
+    WT_RET(__wt_strndup(
+      session, meta_key, key->size, &import_list->entries[import_list->entries_next].name));
+    WT_RET(__wt_strndup(
+      session, meta_value, value->size, &import_list->entries[import_list->entries_next].config));
+    import_list->entries_next++;
+
+    return (0);
+}
+
+/*
+ * __create_parse_export --
+ *     Parse export metadata file and populate array of name/config entries related to uri. The
+ *     array is sorted by entry name. Caller is responsible to free any memory allocated for the
+ *     import list.
+ */
+static int
+__create_parse_export(
+  WT_SESSION_IMPL *session, const char *export_file, WT_IMPORT_LIST *import_list)
+{
+    bool exist;
+
+    exist = false;
+
+    /* Open the specified metadata file and iterate over the key value pairs. */
+    WT_RET(__wt_read_metadata_file(
+      session, export_file, __create_meta_entry_worker, import_list, &exist));
+    if (!exist)
+        return (0);
+
+    /* Sort the array by name. We will use binary search later to get config string. */
```

Comment:
I looked through the entire diff for `import_list` and I'm not seeing where we search or use the list to get the config string. Is that code written and I'm not seeing it or missing? Does it already exist in the current tree so it will just get used automatically?

Reply:
This work will be done in the next ticket [WT-9053](https://jira.mongodb.org/browse/WT-9053)

Reply:
Thanks for clarifying that.